apis/v1beta1/validation/gateway.go

```diff
@@ -91,3 +92,18 @@ func validateListenerHostname(listeners []gatewayv1b1.Listener, path *field.Path
 	}
 	return errs
 }
+
+// validateTLSCertificateRefs validates the certificateRefs
+// must be set when tls config is set and TLSModeType is
+// terminate
+func validateTLSCertificateRefs(listeners []gatewayv1b1.Listener, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	for i, h := range listeners {
+		if h.TLS != nil {
+			if h.TLS.Mode != nil && *h.TLS.Mode == "Terminate" && h.TLS.CertificateRefs == nil {
```

Comment:
You can use `TLSModeTerminate` instead of hardcoding "Terminate". Also, `TLS.Mode` defaults to `Terminate`, therefore you can avoid checking that it is different from nil (it will never be `nil`)

Reply:
Yes, it's correct. Removed it.